evaluators/rank/NdcgEvaluator/__init__.py

```diff
@@ -0,0 +1,43 @@
+from math import log
+from typing import Sequence, Any
+
+from jina.executors.evaluators.rank import BaseRankingEvaluator
+
+
+class NDCGEvaluator(BaseRankingEvaluator):
+    """
+    :class:`NDCGEvaluator` evaluates normalized discounted cumulative gain for information retrieval.
+    """
+
+    @property
+    def metric(self):
+        return f'nDCG@{self.eval_at}'
+
+    def evaluate(self, actual: Sequence[Any], desired: Sequence[Any], *args, **kwargs) -> float:
+        """"
+        :param actual: the scores predicted by the search system.
+        :param desired: the expected score given by user as groundtruth.
+        :return the evaluation metric value for the request document.
+        """
+        actual_at_k = actual[:self.eval_at]
+        desired_at_k = actual[:self.eval_at]
+        if len(actual) < 2:
+            raise ValueError(f'Expecting gains with minimal length of 2, {len(actual)} received.')
+        dcg = self._compute_dcg(gains=actual_at_k)
+        idcg = self._compute_idcg(gains=desired_at_k)
+        if idcg == 0.0:
+            return 0.0
+        else:
+            return dcg/idcg
```

Comment:
```suggestion
        return 0.0 if idcg == 0.0 else dcg / idcg
```

Reply:
it's my personal hobbit to not write if else in one line, but change it also fine